src/accsr/remote_storage.py

```diff
@@ -308,142 +385,82 @@ def _get_push_remote_path(self, local_path: str) -> str:
-
-        remote_obj = [
-            obj
-            for obj in self.bucket.list_objects(remote_path)
-            if not self._listed_due_to_name_collision(remote_path, obj)
-        ]
-        if len(remote_obj) > 1:
-            raise RuntimeError(
-                f"Remote path {remote_path} exists and is a directory, will not overwrite it."
-                f"Consider calling push_directory or push instead."
+        if path_regex is not None and not path_regex.match(path):
+            log.warning(
+                f"{path} does not match regular expression '{path_regex}'. Nothing is pushed."
             )
+            return {"new_files": [], "existing_eq_md5": [], "existing_neq_,d5": []}
+
+        valid_files = []
+        existing_eq_md5 = []
+        existing_neq_md5 = []
+        for file in files:
+            valid = True
+            remote_path = self._get_push_remote_path(file)
+            remote_obj = [
+                obj
+                for obj in self.bucket.list_objects(remote_path)
+                if not self._listed_due_to_name_collision(remote_path, obj)
+            ]
+            if len(remote_obj) > 1:
+                valid = False
+
+            if remote_obj:
+                remote_obj = remote_obj[0]
+                # Skip upload if MD5 hashes match
+                if md5sum(local_path) == remote_obj.hash:
+                    valid = False
+                    existing_eq_md5.append(LocalObject(file))
+                else:
+                    valid = False
+                    existing_neq_md5.append(LocalObject(file))
+
+            if valid:
+                valid_files.append(LocalObject(file))
+
+        summary = {
+            "new_files": valid_files,
+            "existing_eq_md5": existing_eq_md5,
+            "existing_neq_d5": existing_neq_md5,
+        }
 
-        if remote_obj and not overwrite_existing:
-            remote_obj = remote_obj[0]
-            # Skip upload if MD5 hashes match
-            if md5sum(local_path) == remote_obj.hash:
-                log.info(f"Files are identical, skipping upload")
-                return remote_obj
-            elif not overwrite_existing:
-                raise RuntimeError(
-                    f"Remote object {remote_path} already exists,\n is not identical to the local file {local_path}\n "
-                    f"and overwrite_existing=False"
-                )
-
-        log.debug(f"Uploading: {local_path} --> {remote_path}")
-        remote_obj = self.bucket.upload_object(
-            local_path, remote_path, verify_hash=False
-        )
-        return remote_obj
+        return summary
 
     def push(
         self,
         path: str,
         local_path_prefix: Optional[str] = None,
         overwrite_existing=True,
+        force=False,
+        dryrun=False,
         path_regex: Pattern = None,
-    ) -> List[RemoteObjectProtocol]:
+    ) -> Union[List[RemoteObjectProtocol], List[Tuple[str, str]]]:
```

Comment:
The annotation has to change once summaries are returned. I don't see how  `List[Tuple[str, str]]]` can be the return type, rather `List[LocalObject]`